app/views/submitted_content/_submitted_files.html.erb

```diff
@@ -1,120 +1,149 @@
-        <div style="padding-bottom:0.6em"><h5>File actions:</h5>
-        <!--<input type="button" onclick="createNewFolder()"     value="Create new folder"    />
-          <input type="button" onclick="copySelectedFile()"    value="Copy selected file"   />
-          <input type="button" onclick="moveSelectedFile()"    value="Move selected file"   />
-          <input type="button" onclick="renameFile()"          value="Rename selected file" />-->
-          <input type="button" onclick="deleteSelectedFile();" value="Delete selected file" />
+        <div style="padding-bottom:0.6em">
+          <h5>File actions:</h5>
+          <input type="button" onclick="deleteSelectedFile();" value="Delete selected file"/>
           <input type="reset" value="Reset">
         </div>
       <% end %>
 
       <table>
-        <tr><td>&nbsp;</td><td><div id="1_myDiv"><%= display_directory_tree(participant, files, !same_team_flag).html_safe %> </div></td></tr>
+        <tr>
+          <td>&nbsp;</td>
+          <td>
+            <div id="1_myDiv">
+              <%= display_directory_tree(participant, files, !same_team_flag).html_safe %>
+            </div>
+          </td>
+        </tr>
       </table>
     <% end %>
   <% end %>
-  
+
+  <!-- local_assigns is a view helper which contains information
+  about local variables in this partial -->
   <% if local_assigns[:submission_check] %>
     <script type="text/javascript">
-      var supported_file_type = ["pdf", "txt", "doc", "ppt", "png", "jpg"];
-      $(document).ready(function(){
-        var map_id = <%= @map.id %>;
-        var round = <%= @current_round %>;
-        // var current_window_timer = null;
-        var time_out_interval = 500; // in milliseconds
-        var filesToUpdate = {};
-        var windowsToClose = new Array();
-        // startTimer();
-        function timeoutAction(timeout_flag){
-          if (timeout_flag) {
-            while (windowsToClose!=null && windowsToClose.length>0){
-              windowsToClose.shift().close();
+        const supported_file_type = ["pdf", "txt", "doc", "ppt", "png", "jpg"];
+
+        $(document).ready(function () {
+            let map_id = <%= @map.id %>;
+            let round = <%= @current_round %>;
+            let filesToUpdate = {};
+            let windowManager = new WindowManager();
+
+            const time_out_interval = 500; // in milliseconds
+
+            function timeoutAction(timeout_flag) {
+                if (timeout_flag) {
+                    windowManager.closeAll();
+                }
+
+                if ($("#restart_file_timing_flag").val() === 'true') {
+                    filesToUpdate.forEach(function (file) {
+                        if (filesToUpdate[file]) {
+                            startTime(map_id, round, file);
+                        }
+                    });
+
+                    $("#restart_file_timing_flag").val('false');
+                }
             }
```

Comment:
Earlier comment refers to this function being moved to the bottom with checkWindowStatus